docker-compose.yml

```diff
@@ -61,13 +62,95 @@ services:
+      GOVUK_NOTIFY_UNPUBLISHED_POST_NOTICE: "${GOVUK_NOTIFY_UNPUBLISHED_POST_NOTICE}"
+      GOVUK_NOTIFY_UNPUBLISH_PROVIDER_ONE_DAY_NOTICE: "${GOVUK_NOTIFY_UNPUBLISH_PROVIDER_ONE_DAY_NOTICE}"
+      GOVUK_NOTIFY_UNPUBLISHED_PROVIDER_NOTICE: "${GOVUK_NOTIFY_UNPUBLISHED_PROVIDER_NOTICE}"
+      AWS_ACCESS_KEY_ID: "${AWS_ACCESS_KEY_ID}"
+      AWS_SECRET_ACCESS_KEY: "${AWS_ACCESS_KEY_ID}"
+      sandbox: "false"
+#    uncomment this if you have a .env file you want to use locally. Docker compose will throw an error if this does not exist.
+#    env_file: .env
+    depends_on:
+      postgres:
+        condition: service_healthy
+      apply:
+        condition: service_started
+  scheduler-unpublish:
+    build:
+      context: .
+      dockerfile: Dockerfile
+      target: scheduled
+      args:
+        - BUILD_MODE=ci
+    command: ['npm', 'run', 'scheduler:unpublish']
+    environment:
+      DATABASE_URL: "postgresql://master:postgrespass@postgres:5432/lists"
+      FORM_RUNNER_URL: "apply:3001"
+      DEBUG: "true"
+      LOG_LEVEL: "info"
+      CI_SMOKE_TEST: "true"
+      LOCATION_SERVICE_INDEX_NAME: fcdo-professional-service-finder
+      GOVUK_NOTIFY_API_KEY: "${GOVUK_NOTIFY_API_KEY}"
+      GOVUK_NOTIFY_PROFESSIONAL_APPLICATION_EMAIL_CONFIRMATION_TEMPLATE_ID: "${GOVUK_NOTIFY_PROFESSIONAL_APPLICATION_EMAIL_CONFIRMATION_TEMPLATE_ID}"
+      GOVUK_NOTIFY_POST_NEW_PROFESSIONAL_APPLICATION_RECEIVED_TEMPLATE_ID: "${GOVUK_NOTIFY_POST_NEW_PROFESSIONAL_APPLICATION_RECEIVED_TEMPLATE_ID}"
+      GOVUK_NOTIFY_DATA_PUBLISHED_TEMPLATE_ID: "${GOVUK_NOTIFY_DATA_PUBLISHED_TEMPLATE_ID}"
+      GOVUK_NOTIFY_AUTHENTICATION_EMAIL_TEMPLATE_ID: "${GOVUK_NOTIFY_AUTHENTICATION_EMAIL_TEMPLATE_ID}"
+      GOVUK_NOTIFY_EDIT_DETAILS_TEMPLATE_ID: "${GOVUK_NOTIFY_EDIT_DETAILS_TEMPLATE_ID}"
+      GOVUK_NOTIFY_ANNUAL_REVIEW_POST_ONE_MONTH_NOTICE": "${GOVUK_NOTIFY_ANNUAL_REVIEW_POST_ONE_MONTH_NOTICE}"
+      GOVUK_NOTIFY_ANNUAL_REVIEW_POST_ONE_WEEK_NOTICE: "${GOVUK_NOTIFY_ANNUAL_REVIEW_POST_ONE_WEEK_NOTICE}"
+      GOVUK_NOTIFY_ANNUAL_REVIEW_POST_ONE_DAY_NOTICE: "${GOVUK_NOTIFY_ANNUAL_REVIEW_POST_ONE_DAY_NOTICE}"
+      GOVUK_NOTIFY_ANNUAL_REVIEW_POST_STARTED: "${GOVUK_NOTIFY_ANNUAL_REVIEW_POST_STARTED}"
+      GOVUK_NOTIFY_ANNUAL_REVIEW_PROVIDER_STARTED: "${GOVUK_NOTIFY_ANNUAL_REVIEW_PROVIDER_STARTED}"
+      GOVUK_NOTIFY_UNPUBLISH_POST_WEEKLY_NOTICE: "${GOVUK_NOTIFY_UNPUBLISH_POST_WEEKLY_NOTICE}"
+      GOVUK_NOTIFY_UNPUBLISH_POST_ONE_DAY_NOTICE: "${GOVUK_NOTIFY_UNPUBLISH_POST_ONE_DAY_NOTICE}"
+      GOVUK_NOTIFY_UNPUBLISHED_POST_NOTICE: "${GOVUK_NOTIFY_UNPUBLISHED_POST_NOTICE}"
+      GOVUK_NOTIFY_UNPUBLISH_PROVIDER_ONE_DAY_NOTICE: "${GOVUK_NOTIFY_UNPUBLISH_PROVIDER_ONE_DAY_NOTICE}"
+      GOVUK_NOTIFY_UNPUBLISHED_PROVIDER_NOTICE: "${GOVUK_NOTIFY_UNPUBLISHED_PROVIDER_NOTICE}"
+      AWS_ACCESS_KEY_ID: "${AWS_ACCESS_KEY_ID}"
+      AWS_SECRET_ACCESS_KEY: "${AWS_ACCESS_KEY_ID}"
+      sandbox: "false"
```

Comment:
Out of curiosity why are the exact same sets of emails used for both containers?

Reply:
It's a good question. I think it's "for ease".. but there are some mechanisms for reusing keys.

for docker-compose.yml
1. You can use files instead of individual key pairs https://docs.docker.com/compose/environment-variables/#the-env-file
2. use yaml fragments https://docs.docker.com/compose/compose-file/#fragments

for kubernetes.. shall have a think the best way to do this..